### PR TITLE
Feature/mvp 003 initial setup configure jest

### PR DESCRIPTION
**Pull request by Okiana**
- [x] Install and configure the latest version of Jest.
- [x] All files that have a `spec.ts` extension must be run.
- [x] Supertest should be installed and configured for integration testing.
- [x] TypeScript type checking should be activated for jest.
- [x]  Create a test and test watch npm scripts.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+    transform: {
+      '^.+\\.tsx?$': 'ts-jest'
+    },
+    testRegex: '(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$',
+    moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
+    collectCoverage: true,
+    mapCoverage: true,
+  };
```
Comment (Teodor92) [inline jest.config.js]: Can we fix the indentation? Plus missing ending line

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+    transform: {
+      '^.+\\.tsx?$': 'ts-jest'
+    },
+    testRegex: '(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$',
```
Comment (Teodor92) [inline jest.config.js]: Run only ts/tsx files. Let's exclude js from the regex

File: package.json [hunk 3]
```diff
@@ -4,10 +4,25 @@
...
   "scripts": {
-    "start":"npm run build && ts-node src/index.ts",
+    "start": "npm run build && ts-node src/index.ts",
     "start:watch": "npm run build:watch",
     "build": "tsc",
-    "build:watch": "nodemon"
+    "build:watch": "nodemon",
+    "lint": "tslint -p \"./tsconfig.json\" \"./**/*.ts\" -e \"node_modules/**\" -t verbose",
+    "test": "jest",
+    "test:watch": "npm test -- --watch"
```
Comment (Teodor92) [inline package.json]: - it's easier to do `jest -- watch`. Plus I recommend disabling the code coverage report here - it slows  down testing and it's annoying.

File: package.json [hunk 4]
```diff
@@ -21,9 +36,15 @@
   },
   "homepage": "https://github.com/QuestionQuest/interview-me-backend#readme",
   "devDependencies": {
+    "@types/jest": "23.3.5",
```
Comment (Teodor92) [inline package.json]: Just a note - you can use `typesync` to easily sync missing type definitions :) Just a recommendation for the future

File: src/test/sampleTest.spec.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { sum } from '../index'
```
Comment (Teodor92) [inline src/test/sampleTest.spec.ts]: We need to discuss what directory we are going to use for testing or if we are going to put the test files where the actual source code is.

File: src/index.ts [hunk 6]
```diff
@@ -1,2 +1,6 @@
-console.log("Beginning :)")
-console.log("Hello world!")
\ No newline at end of file
+console.log('Beginning :)');
+console.log('Hello world!');
+
+let sum = (a:number, b:number) => a + b 
+
+export { sum }
```
Comment (Teodor92) [inline src/index.ts]: No need to export it like this. You can do:

```typescript
export const sum = (a:number, b:number) => a + b;
```

File: src/test/sampleTest.spec.ts (hunk 5 above)
Comment (Okiana) [reply src/test/sampleTest.spec.ts]: you wanted a simple test. :) This is a simple test

File: package.json (hunk 3 above)
Comment (Okiana) [reply package.json]: I like to see the code coverage report. It's not that slowing down :)

**Pull request merged**